pingcap/pd
# placement: use id as store key

Signed-off-by: disksing <i@disksing.com>

### What problem does this PR solve? 
Fix https://github.com/pingcap/pd/issues/2203

### What is changed and how it works?
- startKey and endKey can be duplicated, use groupID+ID instead
- check data consistency and recover when load rules

### Check List 

Tests 
 - Unit test
 - Manual test (add detailed scripts or steps below)
    - insert some rules with same key, make sure it won't be over written
    - use old pd to write some rules, then start new pd, the wrong key are recovered

Code changes
 - Has persistent data change

Related changes
 - Need to cherry-pick to the release branch
